xstate-chatbot/nodejs/src/machine/hospital-details.js

```diff
@@ -0,0 +1,1016 @@
+    },//Nodal Officer 
+    hospitalTypeId: {
+      id: 'hospitalTypeId',
+      initial: 'process',
+      states: {
+        prompt: {
+          onEntry: assign((context, event) => {
+            let message;
+            if (context.slots.previoushospitaldata.hospital_level_Id == '1') 
+              message = dialog.get_message(messages.L2L3hospitalDetails, context.user.locale);
+            else
+              if (context.slots.previoushospitaldata.hospital_level_Id == '2') 
+               message = dialog.get_message(messages.L2hospitalDetails, context.user.locale);
+            else
+              if (context.slots.previoushospitaldata.hospital_level_Id == '3') 
+              message = dialog.get_message(messages.L3hospitalDetails, context.user.locale);
+              
+              dialog.sendMessage(context, message_3);
+               
+          }),
+          on: {
+            USER_MESSAGE: 'process',
+          },
+        },
+        process: {
+          onEntry: assign((context, event) => {
+            //context.slots.hospital.mobileNumber=context.user.mobileNumber;
+            if (context.slots.previoushospitaldata.hospital_level_Id == '1') {
+              let message = dialog.get_message(messages.L2L3hospitalDetails, context.user.locale);
+              dialog.sendMessage(context, message);
+              context.validMessage = true;
+            }
+            else
+              if (context.slots.previoushospitaldata.hospital_level_Id == '2') {
+                let message_2 = dialog.get_message(messages.L2hospitalDetails, context.user.locale);
+                dialog.sendMessage(context, message_2);
+                context.isValid = false;
+
+              }
+              else
+                if (context.slots.previoushospitaldata.hospital_level_Id == '3') {
+                  let message_3 = dialog.get_message(messages.L3hospitalDetails, context.user.locale);
+                  dialog.sendMessage(context, message_3);
+                  context.isValid = true;
+                }
+          }),
+          always: [
+            {
+              cond: (context) => context.isValid == false,
+              target: '#l2Hospital',
+            },
+            {
+              cond: (context) => context.isValid == true,
+              target: '#l3Hospital',
+            },
+            {
+              cond: (context) => context.validMessage == true,
+              target: '#l2l3Hospital',
+            },
+            {
+              target: 'error',
+            },
+          ],
+        },
+        error: {
+          onEntry: assign((context, event) => {
+            dialog.sendMessage(context, dialog.get_message(messages.noUserFetch, context.user.locale), false);
+          }),
+          always: 'prompt',
+        },
+      },
+    },//Hospital Type Id 1/2/3 
+    l2Hospital: {
+      id: 'l2Hospital',
+      initial: 'process',
+      states: {
+      process: {
+          onEntry: assign((context, event) => {
```

Comment:
There isn't any need to create a substate process. This onEntry function can e part of l2Hospital state.

Reply:
in L2hospital there is further requirement to ask that whether nodal officer wants to update or just want to save old details.